packages/dialtone-vue3/components/mode_island/mode_island.vue

```diff
@@ -0,0 +1,212 @@
+    isInverted () {
+      return this.mode === DT_MODE_ISLAND_TYPES.INVERTED;
+    },
+
+    invertedAttribute () {
+      return this.isInverted ? '' : null;
+    },
+
+    computedMode () {
+      // If mode is explicitly light or dark, use it directly
+      if (this.mode === DT_MODE_ISLAND_TYPES.LIGHT || this.mode === DT_MODE_ISLAND_TYPES.DARK) {
+        return this.mode;
+      }
+
+      // If mode is inverted, use the calculated mode if available
+      if (this.mode === DT_MODE_ISLAND_TYPES.INVERTED && this.calculatedMode) {
+        return this.calculatedMode;
+      }
+
+      // Otherwise calculate it now
+      return this.calculateInvertedMode();
+    },
+  },
+
+  beforeCreate () {
+    // Throw error if data-dt-brand is manually set
+    if (this.$attrs && this.$attrs['data-dt-brand']) {
+      throw new Error('[DtModeIsland] The data-dt-brand attribute is not allowed on mode islands. Brand should only be set at the root level.');
+    }
+  },
+
+  mounted () {
+    // Store element reference
+    this.elementRef = this.$el;
+
+    // Setup MutationObserver to watch for contrast changes on root
+    this.setupContrastObserver();
+
+    // Setup MutationObserver to watch for mode changes (only if inverted)
+    if (this.isInverted) {
+      // Initialize the calculated mode
+      this.calculatedMode = this.calculateInvertedMode();
+      this.setupModeObserver();
+    }
+
+    // Initial contrast value
+    this.currentContrast = getRootContrast();
+  },
+
+  beforeUnmount () {
+    // Cleanup observers
+    if (this.contrastObserver) {
+      this.contrastObserver.disconnect();
+      this.contrastObserver = null;
+    }
+    if (this.modeObserver) {
+      this.modeObserver.disconnect();
+      this.modeObserver = null;
+    }
+  },
+
+  methods: {
+    calculateInvertedMode () {
+      // First check if there's a parent mode island
+      if (this.parentModeIslandMode) {
+        const parentMode = typeof this.parentModeIslandMode === 'function'
+          ? this.parentModeIslandMode()
+          : this.parentModeIslandMode;
+        return getOppositeMode(parentMode);
+      }
+
+      // Otherwise, find the nearest parent with data-dt-mode
+      if (this.elementRef) {
+        const parentMode = findParentMode(this.elementRef);
+        return getOppositeMode(parentMode);
+      }
+
+      // Default fallback
+      return getOppositeMode(getRootMode());
+    },
+
+    setupContrastObserver () {
+      const config = {
+        attributes: true,
+        attributeFilter: ['data-dt-contrast'],
+      };
+
+      const callback = (mutationsList) => {
+        for (const mutation of mutationsList) {
+          if (mutation.type === 'attributes' && mutation.attributeName === 'data-dt-contrast') {
+            this.currentContrast = getRootContrast();
+          }
+        }
+      };
+
+      this.contrastObserver = new MutationObserver(callback);
+      this.contrastObserver.observe(document.documentElement, config);
```

Comment:
Replace it with:

```
this.contrastObserver = new MutationObserver((mutationsList) => {
        for (const mutation of mutationsList) {
          if (mutation.type === 'attributes' && mutation.attributeName === 'data-dt-contrast') {
            this.currentContrast = getRootContrast();
          }
        }
      });

      this.contrastObserver.observe(document.documentElement, {
        attributes: true,
        attributeFilter: ['data-dt-contrast'],
      });
```

Reply:
Address in https://github.com/dialpad/dialtone/pull/945/commits/8a18c9de9df6873f54b94bf49f143fca285ecb61 and https://github.com/dialpad/dialtone/pull/945/commits/a668471a4f9c2e08a6dd320cf95f2f2375e7d32c